builtin/common/misc_helpers.lua

```diff
@@ -458,15 +458,28 @@ do
 end
 
 --------------------------------------------------------------------------------
-function table.copy(t, seen)
-	local n = {}
-	seen = seen or {}
-	seen[t] = n
-	for k, v in pairs(t) do
-		n[(type(k) == "table" and (seen[k] or table.copy(k, seen))) or k] =
-			(type(v) == "table" and (seen[v] or table.copy(v, seen))) or v
-	end
-	return n
+
+function table.copy(t, preserve_metatables)
+	local seen = {}
+	local function copy(value)
+		if type(value) ~= "table" then
+			return value
+		end
+		local t = value
```

Comment:
Minor nitpick: the `t` here shadows the `t` argument from the `table.copy` function.

(That said, the different meanings/uses of `t` are quite easy to distinguish.)